src/main/java/random_csapatnev/MainFrame.java

```diff
@@ -593,59 +625,53 @@ public void gearPickerFrame() {
 		jf.setSize(250, 350);
 		jf.setLayout(new FlowLayout());
 
-		ArrayList<JButton> buttList = new ArrayList<>();
+		addGearButtons(jf, v.gears, false);
+		addGearButtons(jf, v.activeGears, true);
 
-		for (Gear s : v.gears) {
-			JButton item = new JButton(s.name.toString());
-			item.addActionListener(new ActionListener() {
+		jf.setResizable(false);
+		jf.setLocationRelativeTo(frame);
+		jf.setDefaultCloseOperation(javax.swing.WindowConstants.DISPOSE_ON_CLOSE);
+		jf.setVisible(true);
+	}
+
+	private void addGearButtons(final JFrame inputFrame, List<Gear> gears, final boolean isActive) {
+		for (final Gear gear : gears) {
+			JButton button = new JButton(gear.name.toString());
+			button.setFont(new Font(StringLiterals.FONT_NAME, Font.BOLD, 25));
+			button.addActionListener(new ActionListener() {
 				@Override
 				public void actionPerformed(ActionEvent e) {
-					Gear g = null;
-					for (Gear ge : v.gears) {
-						if (ge.name.equals(stringToGearEnum(e.getActionCommand()))) {
-							g = ge;
-						}
-					}
-					if (g != null) {
-						v.equipGear(g.name);
-					}
-					refreshView();
-					jf.dispose();
+					handleGearAction(inputFrame, gear, isActive);
 				}
 			});
-			item.setFont(new Font(StringLiterals.FONT_NAME, Font.BOLD, 25));
-			jf.add(item);
-			buttList.add(item);
+			if (isActive) {
+				button.setForeground(Color.RED);
+			}
+			inputFrame.add(button);
 		}
+	}
 
-		for (Gear s : v.activeGears) {
-			JButton item = new JButton(s.name.toString());
-			item.addActionListener(new ActionListener() {
-				@Override
-				public void actionPerformed(ActionEvent e) {
-					Gear g = null;
-					for (Gear ge : v.activeGears) {
-						if (ge.name.equals(stringToGearEnum(e.getActionCommand()))) {
-							g = ge;
-						}
-					}
-					if (g != null) {
-						v.unequipGear(g.name);
-					}
-					refreshView();
-					jf.dispose();
-				}
-			});
-			item.setFont(new Font(StringLiterals.FONT_NAME, Font.BOLD, 25));
-			item.setForeground(Color.RED);
-			jf.add(item);
-			buttList.add(item);
+	private void handleGearAction(JFrame inputFrame, Gear gear, boolean isActive) {
+		List<Gear> gearList = isActive ? v.activeGears : v.gears;
+		Gear selectedGear = findGearByName(gearList, gear.name);
+		if (selectedGear != null) {
+			if (isActive) {
+				v.unequipGear(selectedGear.name);
+			} else {
+				v.equipGear(selectedGear.name);
+			}
 		}
+		refreshView();
+		inputFrame.dispose();
+	}
 
-		jf.setResizable(false);
-		jf.setLocationRelativeTo(frame);
-		jf.setDefaultCloseOperation(javax.swing.WindowConstants.DISPOSE_ON_CLOSE);
-		jf.setVisible(true);
+	private Gear findGearByName(List<Gear> gearList, GearEnum gearName) {
+		for (Gear gear : gearList) {
+			if (gear.name == gearName) {
```

Comment:
.Equals talan szebb, illetve emlekeim szerint SonarCloud is szol miatta

Reply:
Nem hiba mivel Enum==Enum feltétel, nem String/Boxed típus.